client/src/modules/inventory/list/modals/search.modal.html

```diff
@@ -66,20 +55,12 @@
             </div>
           </div>
 
-          <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.type_id.$invalid }">
-            <label class="control-label" translate>FORM.LABELS.TYPE</label>
+          <bh-inventory-type-select
+            type-id="ModalCtrl.searchQueries.type_id"
+            name="type_id"
```

Comment:
It doesn't look like you've defined the `name` on the component.  You probably can do without it.